src/service/resource.js

```diff
@@ -242,6 +242,7 @@ export class Resource {
     } else {
       this.state_ = ResourceState.NOT_LAID_OUT;
     }
+    this.element.dispatchCustomEvent('amp:built');
```

Comment:
In that case we would need to move `this.state_ = ResourceState.##` to `custom-element.js` as well. I need the state to be changed first before dispatch that customEvent